shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/RouteCardDataTest.kt

```diff
@@ -638,10 +662,1577 @@ class RouteCardDataTest {
+                            listOf(
+                                Alert.InformedEntity.Activity.Board,
+                                Alert.InformedEntity.Activity.Exit,
+                                Alert.InformedEntity.Activity.Ride
+                            ),
+                            route = orangeRoute.id,
+                            routeType = RouteType.HEAVY_RAIL,
+                            stop = northStation.id
+                        ),
+                        Alert.InformedEntity(
+                            activities =
+                            listOf(
+                                Alert.InformedEntity.Activity.Board,
+                                Alert.InformedEntity.Activity.Exit,
+                                Alert.InformedEntity.Activity.Ride
+                            ),
+                            route = orangeRoute.id,
+                            routeType = RouteType.HEAVY_RAIL,
+                            stop = oakGrove.id
+                        ),
+                        Alert.InformedEntity(
+                            activities =
+                            listOf(
+                                Alert.InformedEntity.Activity.Board,
+                                Alert.InformedEntity.Activity.Exit,
+                                Alert.InformedEntity.Activity.Ride
+                            ),
+                            route = orangeRoute.id,
+                            routeType = RouteType.HEAVY_RAIL,
+                            stop = oakGrovePlatform.id
+                        )
+                    )
+            }
+
+        val global = GlobalResponse(objects )
+
+        assertEquals(
+            listOf(RouteCardData(
+                lineOrRoute = RouteCardData.LineOrRoute.Route(orangeRoute),
+                stopData = listOf(
+                    RouteCardData.RouteStopData(northStation, emptyList(), listOf(RouteCardData.Leaf(
+                        directionId = 0,
+                        routePatterns = listOf(orangeSouthboundTypical, orangeSouthboundDiversion),
+                        stopIds = setOf(northStation.id, northStationSouthboundPlatform.id, northStationNorthboundPlatform.id),
+                        upcomingTrips = listOf(           UpcomingTrip(
+                            trip = orangeSouthboundDiversionTrip,
+                            schedule = southboundSchedule
+                        )),
+                        allDataLoaded = true,
+                        // TODO: incorporate alert
+                        alertsHere = emptyList()
+
+                    ),RouteCardData.Leaf(
+                        directionId = 1,
+                        routePatterns = listOf(orangeNorthboundTypical, orangeNorthboundDiversion),
+                        stopIds = setOf(northStation.id, northStationSouthboundPlatform.id, northStationNorthboundPlatform.id),
+                        upcomingTrips = listOf(),
```

Comment:
This is a change from before: we now don't include the arrival-only trips in the list of upcoming trips, but we do still keep the leaf so that we can show the alert.